extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java

```diff
@@ -166,6 +166,19 @@ public class InfinispanClientRuntimeConfig {
     @ConfigItem
     Optional<List<String>> sslCiphers;
 
+    /**
+     * Do SSL hostname validation.
+     * Defaults to true.
+     */
+    @ConfigItem
+    Optional<Boolean> sslHostNameValidation;
```

Comment:
It's unclear to me, why exactly is this optional when no default value is set via SmallRye Config. When no default value is set, then value is either true of false, no? I think setting default value would be clearer.

Reply:
I'm not sure I should write this, because **I don't think it is important at all**, I just thought you could think about it when refactoring stuff in the future:

- you document default to `true`, therefore whether it is default in Infinispan or not, it must be default in Quarkus, therefore saying _I only set this value when user does_ means:
  - you are using Optional when not necessary, you can't just delete default value here because there is none
  - you are documenting default value in description instead of default value
  - boolean is either true or false, so not setting it to the Infinispan client later means you trust you will keep consistency and not forget when anything changes. you are not gaining anything by not setting it, just risking future inconsistency. That is - writing description and having default somewhere else than in Quarkus.